File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int v : nums){
            map.put(v, map.getOrDefault(v, 0)+1);
        }

        for(int v : nums){
            if(map.get(v)>n/2) return v;
        }
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n=len(nums)
        freq = {}
        for num in nums:
            freq[num] = freq.get(num,0)+1
        for key, value in freq.items():
            if value>n//2:
                return key
        return -1
        